### PR TITLE
chore(ui): migrate ContentAreaToolbar component to Typescript and improve ContentContainer and ContentHeading

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

- Migrated `ContentAreaToolbar` to TypeScript, enhancing code quality, maintainability, and scalability.
- Utilised static typing to minimize runtime errors and improve the development experience.
- Improved types of already converted `ContentContainer` and `ContentHeading`.
- Moved deprecated `ContentArea` and `ContentAreaWrapper` to the `/deprecated_js` folder (for deletion when ready) and added it to deprecated section in `Storybook`.

# Changes Made

**TypeScript Migration**:

- Converted `ContentAreaToolbar` component files to TypeScript.
- Added comprehensive type annotations.
- Exported migrated `ContentAreaToolbar`, `ContentArea` and `ContentAreaWrapper` components instead of deprecated ones.
- Moved deprecated `ContentArea` and `ContentAreaWrapper` component to `/deprecated_js` so they can be deleted when ready.

**Testing**:

- Added edge cases:
  - `ContentAreaToolbar`
    - Renders without any props
    - Renders multiple children elements
    - Renders with no children
  - `ContentContainer`
    - Renders without any props
    - Renders multiple children elements
    - Renders with no children
  - `ContentHeading`
    - Renders without any text
    - Render with multiple children elements
    - Renders with no children
- Organised tests by what they check for better readability and maintenance.
- Verified components behaviour using `Storybook`.
- Updated tests to use `vitest`.

**Documentation:**

- Updated `Storybook` documentation.

**Release:**

- Generated a `changeset` file.

# Related Issues

- Closes https://github.com/cloudoperators/juno/issues/221

# Testing Instructions

1. Check `Storybook`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Review by andypf — DISMISSED** [review]
LGTM

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx [hunk 1]
```diff
@@ -20,14 +23,4 @@ const Template = (args: ContentHeadingProps) => <ContentHeading {...args} />
...
-  parameters: {
-    docs: {
-      description: {
-        story: "The the main heading of the content area of a page/view.",
-      },
-    },
-  },
-  args: {
-    heading: "My Page Heading",
-  },
```
Comment (taymoor89) [inline packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx:32]: In the preview there is no default text for the heading.
Screenshot 2024-11-29 at 09 56 26

**Review by taymoor89 — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx (hunk 1 above)
Comment (guoda-puidokaite) [reply packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx:32]: Ooops! I should check after last minute quick changes. 🤪 Thank Youuu! Done.

**Review by taymoor89 — APPROVED** [review]

**Comment by guoda-puidokaite** [comment]
Thank You for the review, @taymoor89. Please re-approve, @andypf.
Let's not merge as @ArtieReus would also like to have a look. But if not reviewed in the afternoon, I'll merge. 💪

**Review by barsukov — APPROVED** [review]

**Pull request merged**